Add makefile to build all images

RJacksonm1 opened the pull request:
Makefile to build images, with options to override organisation or version:

To build the release hardcoded in this repo (v2.4.0), tagged as `thelounge/lounge:v2.4.0[-slim|-alpine]:

```
➜  docker-lounge git:(makefile) ✗ make
...
➜  docker-lounge git:(makefile) ✗ docker images | grep thelounge
thelounge/lounge                       v2.4.0-slim          1b15577445f8        29 seconds ago       242MB
thelounge/lounge                       v2.4.0-alpine        afd450110ea5        55 seconds ago       79.6MB
thelounge/lounge                       v2.4.0               e8b0794da80f        About a minute ago   718MB
```

To build v2.5.0-rc.2:

```
➜  docker-lounge git:(makefile) ✗ THELOUNGE_VERSION=2.5.0-rc.2 make
...
➜  docker-lounge git:(makefile) ✗ docker images | grep thelounge
thelounge/lounge                       v2.5.0-rc.2-alpine   bd4b85402d38        6 hours ago         80MB
thelounge/lounge                       v2.5.0-rc.2-slim     7388a4e06ead        6 hours ago         242MB
thelounge/lounge                       v2.5.0-rc.2          f4ee3aeaa52e        6 hours ago         718MB

```

And to tag it to my own org:

```
➜  docker-lounge git:(makefile) ✗ ORGANISATION=rjacksonm1 THELOUNGE_VERSION=2.5.0-rc.2 make
...
➜  docker-lounge git:(makefile) ✗ docker images | grep rjacksonm1/lounge
rjacksonm1/lounge                      v2.5.0-rc.2-alpine   bd4b85402d38        6 hours ago         80MB
rjacksonm1/lounge                      v2.5.0-rc.2-slim     7388a4e06ead        6 hours ago         242MB
rjacksonm1/lounge                      v2.5.0-rc.2          f4ee3aeaa52e        6 hours ago         718MB
```

YaManicKill commented:
So, should build run all 3, or should it just default to the main one, and have another command to run all 3?

RJacksonm1 commented:
`make` will build all three (equivalent to `make build`; makefiles will default to the first target defined).

Otherwise, we can build individual images with `make build-main`, `make build-alpine`, ...

Might be nicer if I changed the targets and removed the `build-` prefix (so `make main`, `make alpine` instead); looks like I can use [`.PHONY`](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html) to achieve this, which I wasn't aware of earlier.

YaManicKill commented:
No, I understand what it does, I'm asking what it should do.

I think that the default should just build the main one, as most people won't be wanting to get all 3.

RJacksonm1 commented:
I think the default action should be to build everything this repository encompasses – so all three.

williamboman commented:
I rarely use make, but isn't one core principle that every make command is reversable (should deleting images be supported somehow)? Also, isn't it convention that the command one issues is also where output will be placed, e.g. `make main` would produce files in `./main`?

To circumvent these conventions one can add the commands to the `.PHONY` list, correct? Shouldn't `main` be added to `.PHONY` then?

RJacksonm1 commented:
@williamboman Aah, I'm not aware of any particular conventions with regards to makefiles; we use them at my place of employment primarily as developer-friendly shortcuts to run a bunch of common commands, which was my intent in bringing it here.

I think your comments make sense if we're talking about `make` in the context of building binary assets, but in the context of docker images... I'm learning toward thinking it doesn't really matter. I am wanting to get up to speed on any makefile conventions now though!

williamboman commented:
> I am wanting to get up to speed on any makefile conventions now though!

I'm not very dogmatic, I'm 👍 as-is. Or do you want to research this and report back later 😄 ?

RJacksonm1 commented:
I probably won't get around to researching anytime soon; I've got too many other things on my plate.

The PR will provide _some_ degree of value as-is, even if it doesn't completely align with whatever the `makefile` conventions might be, so I'd give it the thumbs up as well.
